disnake/guild.py

```diff
@@ -3311,6 +3312,89 @@ async def query_members(
             self, query=query, limit=limit, user_ids=user_ids, presences=presences, cache=cache
         )
 
+    async def get_or_fetch_members(
+        self,
+        member_ids: List[int],
+        *,
+        presences: bool = False,
+        cache: bool = True,
+    ) -> List[Member]:
+        """|coro|
+
+        Tries to get the guild members matching the provided IDs from cache.
+        If some of them were not found, the method requests the missing members using websocket operations.
+        If ``cache`` kwarg is ``True`` (default value) the missing members will be cached.
+
+        If more than 100 members are missing, several websocket operations are made. Otherwise only one.
+
+        Websocket operations can be slow, however, this method is cheaper than multiple :meth:`get_or_fetch_member` calls.
+
+        .. versionadded:: 2.4
+
+        Parameters
+        -----------
+        member_ids: List[:class:`int`]
+            List of user IDs to search for. If the user ID is not in the guild then it won't be returned.
+        presences: :class:`bool`
+            Whether to request for presences to be provided. This defaults to ``False``.
+        cache: :class:`bool`
+            Whether to cache the missing members internally. This makes operations
+            such as :meth:`get_member` work for those that matched.
+            It also speeds up this method on repeated calls. Defaults to ``True``.
+
+        Raises
+        -------
+        asyncio.TimeoutError
+            The query timed out waiting for the members.
+        ClientException
+            The presences intent is not enabled.
+
+        Returns
+        --------
+        List[:class:`Member`]
+            The list of members that have matched the IDs.
+        """
+        if presences and not self._state._intents.presences:
+            raise ClientException("Intents.presences must be enabled to use this.")
+
+        members: List[Member] = []
+        unresolved_ids: List[int] = []
+
+        for member_id in member_ids:
+            member = self.get_member(member_id)
+            if member is None:
+                unresolved_ids.append(member_id)
+            else:
+                members.append(member)
+
+        if not unresolved_ids:
+            return members
+
+        if len(unresolved_ids) == 1:
+            # fetch_member is cheaper than query_members
+            try:
+                members.append(await self.fetch_member(unresolved_ids[0]))
+            except HTTPException:
+                pass
+        else:
+            # We have to split the request into several smaller requests
+            # because the limit is 100 members per request.
+            # Imo no one should ever request more than 100 members using this method.
```

Comment:
why not?

Reply:
I can't think of a use case where you need to fetch more than 100 members. People usually fetch members to get full tags in order to display them in a nice leaderboard or something like that, so 10-20 users should be enough. But that's just my opinion based on intuition. I will remove this line, it's confusing.